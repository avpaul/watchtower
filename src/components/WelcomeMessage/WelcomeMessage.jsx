import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import getTimeMessage from '../../utils/getTimeMessage';
import './index.scss';

const WelcomeMessage = props => {
  const { user } = props;
  const currentHour = parseFloat(moment().format('HH.mm'));

  const message = getTimeMessage(currentHour, user.first_name);

  return (
    <div>
      <span className="welcome-text">{message}</span>
    </div>
  );
};

WelcomeMessage.propTypes = {
  user: PropTypes.shape({}).isRequired
};

export default WelcomeMessage;
