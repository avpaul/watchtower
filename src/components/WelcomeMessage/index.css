.welcome-text {
  height: 31px;
  width: 167px;
  color: #344c5a;
  font-family: 'DIN Pro';
  font-size: 24px;
  font-weight: 500;
  letter-spacing: -0.69px;
  line-height: 31px;
}
