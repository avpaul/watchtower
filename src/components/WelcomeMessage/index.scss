@import '../../assets/sass/abstract/variables';

.welcome-text {
  height: 31px;
  width: 167px;
  color: #344c5a;
  font-family: $font-primary;
  font-size: 24px;
  font-weight: 500;
  letter-spacing: -0.69px;
  line-height: 31px;
}
