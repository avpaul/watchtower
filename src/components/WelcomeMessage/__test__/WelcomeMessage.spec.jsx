import React from 'react';
import { shallow } from 'enzyme';
import WelcomeMessage from '../WelcomeMessage';

jest.mock('moment', () => () => ({ format: () => 16.00 }));

const props = {
  user: {
    first_name: 'Erick'
  }
};

it('matches snapshot', () => {
  const tree = shallow(<WelcomeMessage {...props} />);
  expect(tree).toMatchSnapshot();
});
