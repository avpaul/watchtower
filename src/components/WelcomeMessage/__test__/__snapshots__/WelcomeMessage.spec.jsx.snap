// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <span
    className="welcome-text"
  >
    Hey Erick, you're doing great! 👍
  </span>
</div>
`;
